### PR TITLE
Remove AutoComponent

### PR DESCRIPTION
**Pull request by McCallisterRomer**
Removed AutoComponent and added FrameLogic, FixedLogic, and CollisionLogic components as replacements.

### CODE REVIEW

File: docs/EngineComponents.md [hunk 1]
```diff
@@ -12,6 +12,50 @@ Engine Components
 * [Camera](#camera)
 * [ParticleEmitter](#particleemitter)
 * [NetworkDispatcher](#networkdispatcher)
+* [FreeComponentGroup](#freecomponentgroup)
+
+### Logic
+---------
+There are three Components designed for quickly hooking up custom logic. They each each hold a callable type which is invoked under specific conditions:
```
Comment (McCallisterRomer) [inline docs/EngineComponents.md]: 'each' repeated

File: nc/include/ecs/HandleManager.h [hunk 2]
```diff
@@ -3,6 +3,7 @@
 #include "Entity.h"
 
 #include <algorithm>
+#include <span>
```
Comment (McCallisterRomer) [inline nc/include/ecs/HandleManager.h:7]: pretty sure iterator will be required

File: nc/include/type/EngineTypes.h [hunk 3]
```diff
@@ -13,7 +13,7 @@ namespace nc
 
     /** @todo Only registering names of these: */
     REGISTER_TYPE(AudioSource);
-    REGISTER_TYPE(AutoComponentGroup);
+    REGISTER_TYPE(FreeComponentGroup);
```
Comment (McCallisterRomer) [inline nc/include/type/EngineTypes.h:16]: Need to add logic types too.

File: nc/source/ecs/component/Component.cpp [hunk 4]
```diff
@@ -5,9 +5,9 @@
 
 namespace nc
 {
-    void AutoComponent::ComponentGuiElement()
+    void FreeComponent::ComponentGuiElement()
     {
-        ImGui::Text("User AutoComponent");
+        ImGui::Text("Unknown FreeComponent");
```
Comment (jaremieromer) [inline nc/source/ecs/component/Component.cpp:10]: When would this be hit?

**Review by jaremieromer — APPROVED** [review]

File: nc/source/ecs/component/Component.cpp (hunk 4 above)
Comment (McCallisterRomer) [reply nc/source/ecs/component/Component.cpp:10]: For any user-defined FreeComponent that doesn't override it. It still shows up in the editor as a component, but we don't know how to present more data than "something exists here."

**Pull request merged**